mozilla-central-patches/fix-search-test-race.diff

```diff
@@ -0,0 +1,21 @@
+diff --git a/browser/components/search/test/browser_searchEngine_behaviors.js b/browser/components/search/test/browser_searchEngine_behaviors.js
+--- a/browser/components/search/test/browser_searchEngine_behaviors.js
++++ b/browser/components/search/test/browser_searchEngine_behaviors.js
+@@ -95,14 +95,9 @@ function promiseStateChangeURI() {
+ 
+ function promiseContentSearchReady(browser) {
+   return ContentTask.spawn(browser, {}, async function(args) {
+-    return new Promise(resolve => {
+-      content.addEventListener("ContentSearchService", function listener(aEvent) {
+-        if (aEvent.detail.type == "State") {
+-          content.removeEventListener("ContentSearchService", listener);
+-          resolve();
+-        }
+-      });
+-    });
++    await ContentTaskUtils.waitForCondition(() => content.wrappedJSObject.gContentSearchController &&
```

Comment:
It'd be more readable if each of the "content.wrappedJSObject" clauses had its own line, eslint-permitting.

Reply:
The original style of test seems more robust, though I get that it wasn't quite right.  This fix seems ok for now, but I'd suggest also adding an XXX comment suggesting adding an event to the code or some such so that the test depends less on implementation details and is more useful for refactoring.

Reply:
I don't really see how this relies more on implementation details, given that that particular event has to be fired (and also in a particular order, i.e. after the listener is added in the test). Additionally, I kind of feel like using the event makes this test way harder to debug/step through 😜

Reply:
OK, that's a reasonable enough rationale.